api/repositories/org_repository.go

```diff
@@ -393,6 +401,91 @@ func (r *OrgRepo) GetSpace(ctx context.Context, info authorization.Info, spaceGU
 	return spaceRecords[0], nil
 }
 
+func (r *OrgRepo) DeleteOrg(ctx context.Context, info authorization.Info, message DeleteOrgMessage) error {
+	// Fetch 'hierarchy' object using the privileged client as we want to return a 'not-found' error for
+	// non-existent orgs as opposed to saying 'permission-denied'.
+	hierarchyObj := v1alpha2.HierarchyConfiguration{}
+	err := r.privilegedClient.Get(ctx, types.NamespacedName{Name: "hierarchy", Namespace: message.GUID}, &hierarchyObj)
```

Comment:
"hierarchy" should be extracted into a constant instead of a magic string.

Reply:
Addressed in commit - https://github.com/cloudfoundry/cf-k8s-controllers/pull/464/commits/a99f88f188236f227f4a094857f8ada66c9204e7